platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/merkle/VirtualRootNode.java

```diff
@@ -323,8 +324,10 @@ public VirtualRootNode() {
     public VirtualRootNode(final @NonNull VirtualDataSourceBuilder<K, V> dataSourceBuilder) {
         this.fastCopyVersion = 0;
         this.hasher = new VirtualHasher<>();
-        this.dataSourceBuilder = enforce ? Objects.requireNonNull(dataSourceBuilder) : dataSourceBuilder;
         this.flushThreshold.set(settings.getCopyFlushThreshold());
+        Objects.requireNonNull(dataSourceBuilder);
```

Comment:
nit: Objects.requireNonNull returns the parameter, so you could combine these two lines.